infrastructure/templates/public-api/application/shared/recoveryVault.bicep

```diff
@@ -0,0 +1,55 @@
+import { ResourceNames } from '../../types.bicep'
+
+@description('Common resource naming variables.')
+param resourceNames ResourceNames
+
+@description('The location to create resources in.')
+param location string
+
+@description('Whether manual deletion of backups is allowed.')
+param immutable bool
+
+@description('Tags for the resources')
+param tagValues object
+
+module recoveryVaultModule '../../components/recoveryVault.bicep' = {
+  name: 'recoveryVaultDeploy'
+  params: {
+    vaultName: resourceNames.sharedResources.recoveryVault
+    location: location
+    redundancy: 'GeoRedundant'
+    immutable: immutable
+    softDelete: {
+      retentionDays: 30
+      state: 'AlwaysON'
+    }
+    crossSubscriptionRestoreEnabled: false
+    crossRegionRestoreEnabled: true
+    alertsEnabled: true
+    tagValues: tagValues
+  }
+}
+
+module fileShareBackupPolicyModule '../../components/recoveryVaultFileShareBackupPolicy.bicep' = {
+  name: 'fileShareBackupPolicyDeploy'
+  params: {
+    policyName: resourceNames.sharedResources.recoveryVaultFileShareBackupPolicy
+    vaultName: resourceNames.sharedResources.recoveryVault
+    location: location
+    dailyBackupTimeUtc: '00:00'
+    daysToRetainDailyBackups: 30
+    weeklySnapshotRetention: {
+      targetBackupDay: 'Monday'
+      weeksToRetain: 8
+    }
+    monthlySnapshotRetention: {
+      targetBackupDay: 'Monday'
+      targetBackupWeek: 'First'
+      monthsToRetain: 12
+    }
```

Comment:
I noticed that a yearly retention schedule isn't defined here but support has been added in elsewhere for configuring one should we need it. I'm guessing that you don't think retaining a yearly backup is necessary?

Tbh, I can't think why we'd need to back that far, and it might only ever be out of curiosity that we'd need to peek into one that old anwyay, rather than to restore from.

Reply:
Yeah exactly that!  I don't normally add in support for EVERY ARM feature to our Bicep templates, but normally limit it to the ones that we're using or are likely to use.  In this case, although I know we're not using it now, it seemed such little effort to add support for yearly too so I just added it in as well for the future!